src/executor/plan_executor.cpp

```diff
@@ -129,7 +130,7 @@ ExecuteResult PlanExecutor::ExecutePlan(const planner::AbstractPlan *plan,
     result.clear();
 
     // Bind: casting const should be removed with later refactoring executor
-    planner::AbstractPlan *planp = const_cast<planner::AbstractPlan *>(plan);
+    planner::AbstractPlan *planp = plan.get();
```

Comment:
We can remove this _planp_ pointer and directly use the _plan_ pointer now.  The previous casting was only necessary to remove _const_.  And it would be nice if you could also change the comment accordingly.